### PR TITLE
Turn all for loops using "begin_active" into ranged-based ones

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

**Comment by bangerth** [comment]
Actually, `grid/grid_out_svg_02.debug` fails.

File: source/grid/grid_out.cc [hunk 1]
```diff
@@ -1398,10 +1383,7 @@ GridOut::write_xfig(const Triangulation<2> &tria,
   // on finer levels. Level 0
   // corresponds to a depth of 900,
   // each level subtracting 1
-  Triangulation<dim, spacedim>::cell_iterator       cell = tria.begin();
-  const Triangulation<dim, spacedim>::cell_iterator end  = tria.end();
-
-  for (; cell != end; ++cell)
+  for (const auto &cell : tria.active_cell_iterators())
```
Comment (bangerth) [inline source/grid/grid_out.cc]: Here's your bug. You should loop over all, not just the active cells.

Comment (masterleinad) [reply source/grid/grid_out.cc]: Thanks!

File: source/multigrid/mg_tools.cc [hunk 2]
```diff
@@ -138,7 +135,7 @@ namespace MGTools
     // unknown number of cells may
     // share an object of dimension
     // smaller than dim-1.
-    for (cell = dofs.begin(level); cell != end; ++cell)
+    for (const auto &cell : dofs.active_cell_iterators_on_level(level))
```
Comment (rezarastak) [inline source/multigrid/mg_tools.cc]:
```suggestion
    for (const auto &cell : dofs.cell_iterators_on_level(level))
```

File: source/multigrid/mg_tools.cc [hunk 3]
```diff
@@ -332,7 +326,7 @@ namespace MGTools
     // unknown number of cells may
     // share an object of dimension
     // smaller than dim-1.
-    for (cell = dofs.begin_active(); cell != end; ++cell)
+    for (const auto &cell : dofs.active_cell_iterators_on_level(level))
```
Comment (rezarastak) [inline source/multigrid/mg_tools.cc]: The original code is confusing, the for loop starts at `dofs.begin_active()` but ends at `dofs.end(level)`.

Comment (tjhei) [reply source/multigrid/mg_tools.cc]: I am not sure that the old code is correct. I would go from begin(level) to end(level).

Comment (masterleinad) [reply source/multigrid/mg_tools.cc]: Hmm... I am not sure either, but for me it sounds more reasonable to loop over active cells on each level if we want to find out about the number of coupling degrees of freedom on a particular level. Unfortunately, there is no test for this function.

Comment (tjhei) [reply source/multigrid/mg_tools.cc]: We are asking for ``get_mg_dof_indices``, so I don't see a reason for looking at active cells only.

Comment (masterleinad) [reply source/multigrid/mg_tools.cc]: Thinking some more about it, I guess you're right. I changed the pull request accordingly.

**Pull request merged**